### PR TITLE
Implemented pumpkins/shears

### PR DESCRIPTION
**Pull request by DaPigGuy**
Added pumpkins, carved pumpkins, jack o lanterns (lit pumpkins), & shears with pumpkin carving

### CODE REVIEW

**Comment by T14Raptor** [comment]
What about wearing a carved pumpkin?

**Comment by Sandertv** [comment]
For allowing the wearing of carved pumpkins, navigate to `item/inventory/armour.go` and edit the `canEditArmour` method. `case 0` should check if the item is a pumpkin if it is not a helmet.

File: dragonfly/item/shear.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/internal/item_internal"
+	"github.com/df-mc/dragonfly/dragonfly/item/tool"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/go-gl/mathgl/mgl64"
+)
+
+// Shear is a tool used to shear sheep, mine a few types of blocks, and carve pumpkins.
+type Shear struct {
```
Comment (Sandertv) [inline dragonfly/item/shear.go:11]: Might be better to name this (and the file) `Shears`.

File: dragonfly/item/shear.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+// BaseMiningEfficiency ...
+func (s Shear) BaseMiningEfficiency(b world.Block) float64 {
+	return 1.5
+}
+
+// DurabilityInfo ...
+func (s Shear) DurabilityInfo() DurabilityInfo {
+	return DurabilityInfo{
+		MaxDurability:    238,
+		BrokenItem:       nil,
```
Comment (Sandertv) [inline dragonfly/item/shear.go:45]: Not sure if this'll work properly, use `simpleItem(Stack{})`` instead.

**Pull request merged**